queries.sql

```diff
@@ -7,4 +7,39 @@ SELECT date_of_birth FROM animals WHERE name ='piachu' OR name ='Agumon';
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered=true;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN '10.4' AND '17.3';
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN '10.4' AND '17.3';
+
+BEGIN;
+  update animals
+  set species = 'unspecified';
+  ROLLBACK;
+
+  BEGIN;
+  update animals
+  set species = 'digimon' where name like '%mon';
+  update animals
+  set species = 'pokemon' where name not like '%mon';
+  COMMIT;
+
+  BEGIN;
+  DELETE from animals;
+--   the ROLLBACK should be after executing the delete
+  ROLLBACK;
+
+  BEGIN;
+  DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+  SAVEPOINT delete_2022;
+  UPDATE animals
+  SET weight_kg = weight_kg * -1;
+  ROLLBACK TO SAVEPOINT delete_2022;
+  UPDATE animals
+  SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+  COMMIT;
+  
+
+SELECT count(*) FROM animals;
+SELECT count(*) FROM animals WHERE escape_attempts > 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, MAX(escape_attempts) FROM animals GROUP BY neutered;
+SELECT species, MIN(weight_kg), MAX(weight_kg) from animals GROUP BY species;
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-30' GROUP BY species;
```

Comment:
* Kindly adjust this query `WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-30'` as any animal born on the 31st of dec 2000 wont be counted so the correct query should be `WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'`